tests/utils/test_utils.go

```diff
@@ -0,0 +1,433 @@
+	}
+
+	if useParser {
+		LogInfoMessage(fmt.Sprintf("Use Parser to write devfile %s", fileName))
+		newDevfile, createErr := devfileData.NewDevfileData(devfile.SchemaDevFile.SchemaVersion)
+		if createErr != nil {
+			err = errors.New(LogErrorMessage(fmt.Sprintf("Creating new devfile : %v", createErr)))
+		} else {
+			newDevfile.SetSchemaVersion(devfile.SchemaDevFile.SchemaVersion)
+
+			// add the commands to new devfile
+			for _, command := range devfile.SchemaDevFile.Commands {
+				newDevfile.AddCommands(command)
+			}
+			// add components to the new devfile
+			newDevfile.AddComponents(devfile.SchemaDevFile.Components)
+
+			ctx := devfileCtx.NewDevfileCtx(fileName)
+
+			err = ctx.SetAbsPath()
+			if err != nil {
+				LogErrorMessage(fmt.Sprintf("Setting devfile path : %v", err))
+			} else {
+				devObj := parser.DevfileObj{
+					Ctx:  ctx,
+					Data: newDevfile,
+				}
+				err = devObj.WriteYamlDevfile()
+				if err != nil {
+					LogErrorMessage(fmt.Sprintf("Writing devfile : %v", err))
+				} else {
+					devfile.SchemaParsed = false
+				}
+			}
+
+		}
+	} else {
+		LogInfoMessage(fmt.Sprintf("Marshall and write devfile %s", devfile.FileName))
+		c, marshallErr := yaml.Marshal(&(devfile.SchemaDevFile))
+
+		if marshallErr != nil {
+			err = errors.New(LogErrorMessage(fmt.Sprintf("Marshall devfile %s : %v", devfile.FileName, marshallErr)))
+		} else {
+			err = ioutil.WriteFile(fileName, c, 0644)
+			if err != nil {
+				LogErrorMessage(fmt.Sprintf("Write devfile %s : %v", devfile.FileName, err))
+			} else {
+				devfile.SchemaParsed = false
+			}
+		}
+	}
+	return err
+}
+
+// Use the parser to parse a devfile on disk
+func (devfile *TestDevfile) parseSchema() error {
+
+	var err error
+	if !devfile.SchemaParsed {
+		LogInfoMessage(fmt.Sprintf("Parse and Validate %s : ", devfile.FileName))
+		devfile.ParsedSchemaObj, err = devfilepkg.ParseAndValidate(devfile.FileName)
+		if err != nil {
+			LogErrorMessage(fmt.Sprintf("From ParseAndValidate %v : ", err))
+		}
+		devfile.SchemaParsed = true
+	}
+	return err
+}
+
+// Verify verifies the contents of the specified devfile with the expected content
+func (devfile TestDevfile) Verify() error {
+
+	LogInfoMessage(fmt.Sprintf("Verify %s : ", devfile.FileName))
+
+	var errorString []string
+
+	err := devfile.parseSchema()
+
+	if err != nil {
+		errorString = append(errorString, LogErrorMessage(fmt.Sprintf("parsing schema %s : %v", devfile.FileName, err)))
+	} else {
+		LogInfoMessage(fmt.Sprintf("Get commands %s : ", devfile.FileName))
+		commands, _ := devfile.ParsedSchemaObj.Data.GetCommands(common.DevfileOptions{})
+		if commands != nil && len(commands) > 0 {
+			err = devfile.VerifyCommands(commands)
+			if err != nil {
+				errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Verfify Commands %s : %v", devfile.FileName, err)))
+			}
+		} else {
+			LogInfoMessage(fmt.Sprintf("No command found in %s : ", devfile.FileName))
+		}
+	}
+
+	if err == nil {
```

Comment:
no point is another if block? You can have the logic in the same block

Reply:
Moved.